_posts/2023/2023-09-11-express-yourself-clearly-with-positive-negative-numbers.md

```diff
@@ -0,0 +1,142 @@
+---
+title: "Express yourself clearly with positive? and negative? for numbers"
+description: "Syntactical sugar to avoid comparison operators"
+layout: article
+category: ruby
+image:
+  base: "2023/express-yourself-clearly-with-positive-negative-numbers"
+  alt: "The architecture of Birmingham, UK"
+  credit: "Mike Hindle"
+  source: "https://unsplash.com/photos/ow87XYsaY6Q"
+
+---
+
+Ruby, in contrast to other languages, often provides multiple ways to accomplish simple programming tasks. In pursuit of developer happiness, the Standard Library offers the opportunity to make your code appear more like English.
+
+In this case we’ll look at comparing numbers with zero.
+
+## Instead of…
+
+…using comparison operators:
+
+```ruby
+if number > 0
+  # do a thing
+end
+
+if number < 0
+  # do a thing
+end
+```
+
+## Use…
+
+…ruby’s convenience methods on `Numeric`, and its subclasses, to express yourself more clearly:
+
+```ruby
+if number.positive?
+  # do a thing
+end
+
+if number.negative?
+  # do a thing
+end
+```
+
+
+## Why?
+
+I’ve previously discussed my preference for a similar comparator, `zero?`, in a [previous article](/ruby/use-zero-for-integer-float-comparison).
+
+The “more English” version to be clearer and easier to reason about when you return to your code in the future. It can require a moment of concentration to unpack the orientation of the angle bracket and how it relates to the numbers and variables.
```

Comment:
*will be clearer

Reply:
```suggestion
The “more English” version will be clearer and easier to reason about when you return to your code in the future. It can require a moment of concentration to unpack the orientation of the angle bracket and how it relates to the numbers and variables.
```